src/main/java/com/minecolonies/coremod/client/gui/GuiMinecoloniesChest.java

```diff
@@ -0,0 +1,59 @@
+package com.minecolonies.coremod.client.gui;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.inventory.ContainerChest;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.util.ResourceLocation;
+
+/**
+ * ------------ Class not Documented ------------
+ */
+public class GuiMinecoloniesChest extends GuiContainer
+{
+    /**
+     * The ResourceLocation containing the chest GUI texture.
+     */
+    private static final ResourceLocation CHEST_GUI_TEXTURE = new ResourceLocation("minecraft:textures/gui/container/generic_54.png");
+    private final IInventory upperChestInventory;
+    private final IInventory lowerChestInventory;
+    /**
+     * window height is calculated with these values; the more rows, the heigher
+     */
+    private final int        inventoryRows;
+
+    public GuiMinecoloniesChest(IInventory upperInv, IInventory lowerInv)
+    {
+        super(new ContainerChest(upperInv, lowerInv, Minecraft.getMinecraft().player));
+        this.upperChestInventory = upperInv;
+        this.lowerChestInventory = lowerInv;
+        this.allowUserInput = false;
+        int i = 222;
+        int j = 114;
+        this.inventoryRows = lowerInv.getSizeInventory() / 9;
+        this.ySize = 114 + this.inventoryRows * 18;
+    }
+
+    /**
+     * Draw the foreground layer for the GuiContainer (everything in front of the items)
+     */
+    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
+    {
+        this.fontRenderer.drawString(this.lowerChestInventory.getDisplayName().getUnformattedText(), 8, 6, 4210752);
+        this.fontRenderer.drawString(this.upperChestInventory.getDisplayName().getUnformattedText(), 8, this.ySize - 96 + 2, 4210752);
+    }
+
+    /**
+     * Draws the background layer of this container (behind the items).
+     */
+    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY)
+    {
+        GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
+        this.mc.getTextureManager().bindTexture(CHEST_GUI_TEXTURE);
+        int i = (this.width - this.xSize) / 2;
+        int j = (this.height - this.ySize) / 2;
+        this.drawTexturedModalRect(i, j, 0, 0, this.xSize, this.inventoryRows * 18 + 17);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 18 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 17 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)